src/main/java/core/basesyntax/service/DataConverterImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            try {
+                String[] elements = line.split(",");
+                if (elements.length != 3) {
+                    throw new IllegalArgumentException("Invalid input format: " + line);
+                }
+                FruitTransaction.Operation operation =
+                        FruitTransaction.Operation.fromCode(elements[0]);
+                String fruit = elements[1];
+                int quantity = Integer.parseInt(elements[2]);
+                transactions.add(new FruitTransaction(operation, fruit, quantity));
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException(
+                        "Invalid quantity in line: " + line, e);
```

Comment:
Consider providing more specific information about the invalid quantity, such as the expected format or range, to help users correct their input.